### PR TITLE
Add search to card source modal in dashboards

### PR DESCRIPTION
**Pull request by ranquild**
Epic https://github.com/metabase/metabase/issues/26898

How to test:
- Follow the steps from https://github.com/metabase/metabase/pull/27354 but search for a question instead of selecting one manually

Screenshot 2022-12-23 at 18 41 06

- - -
This change is [Reviewable](https://reviewable.io/reviews/metabase/metabase/27405)

### CODE REVIEW

File: frontend/test/metabase/scenarios/dashboard-filters/dashboard-filters-source.cy.spec.js [hunk 1]
```diff
@@ -20,7 +20,7 @@ const dashboardQuestionDetails = {
 };
 
 const structuredQuestionDetails = {
-  name: "GUI categories",
+  name: "Categories",
```
Comment (WiNloSt) [inline frontend/test/metabase/scenarios/dashboard-filters/dashboard-filters-source.cy.spec.js:23]: Not quite sure what's the reason for changing both question names to the same name. I think the previous name is more explicit and easier to understand when reading each test individually.

File: frontend/test/metabase/scenarios/dashboard-filters/dashboard-filters-source.cy.spec.js [hunk 2]
```diff
@@ -94,8 +94,8 @@ describe("scenarios > dashboard > filters", () => {
 const setupStructuredQuestionSource = () => {
   cy.findByText("Values from a model or question").click();
   modal().within(() => {
-    cy.findByText("Saved Questions").click();
-    cy.findByText("GUI categories").click();
+    cy.findByPlaceholderText(/Search for a question/).type("Categories");
```
Comment (WiNloSt) [inline frontend/test/metabase/scenarios/dashboard-filters/dashboard-filters-source.cy.spec.js:97]: Should we have another case that could return results not found?
e.g.
image

File: frontend/src/metabase/parameters/components/CardValuesSourceModal/CardStepModal.tsx [hunk 3]
```diff
@@ -75,13 +80,40 @@ const CardStepModal = ({
...
+        <DataPickerContainer>
+          <DataPicker value={value} onChange={setValue} />
+        </DataPickerContainer>
+      </DataPicker.Provider>
     </ModalContent>
   );
 };
 
+const DataPickerSearchInput = () => {
+  const { search } = useDataPicker();
```
Comment (WiNloSt) [inline frontend/src/metabase/parameters/components/CardValuesSourceModal/CardStepModal.tsx:94]: Cool! Seems like the `DataPickerContainer` is already doing the heavy-lifting searching questions for us.

**Review by WiNloSt — COMMENTED** [review]
I think we should add another test when the search result returns nothing.

File: frontend/test/metabase/scenarios/dashboard-filters/dashboard-filters-source.cy.spec.js (hunk 2 above)
Comment (ranquild) [reply frontend/test/metabase/scenarios/dashboard-filters/dashboard-filters-source.cy.spec.js:97]: I believe this should be tested on the picker level and not by us. This looks like an issue with the underlying component.

Comment (WiNloSt) [reply frontend/test/metabase/scenarios/dashboard-filters/dashboard-filters-source.cy.spec.js:97]: Sounds good. Do we have tests for this for DataPicker component, if not should we create new ones?

Comment (ranquild) [reply frontend/test/metabase/scenarios/dashboard-filters/dashboard-filters-source.cy.spec.js:97]: @WiNloSt currently @kulyk is working on adding tests for the picker

**Review by WiNloSt — APPROVED** [review]
Works great.

**Pull request merged**